
### 파이썬 for 문은 횟수가 정해져 있거나 상대적으로 횟수가 적은 경우에 대한 반복에 주로 사용됩니다. 파이썬 while문은 횟수가 상대적으로 크거나 무한 루프 형태의 반복문에 사용됩니다.

#### 131. for문의 실행결과를 예측하라.

> 과일 = ["사과", "귤", "수박"]

> for 변수 in 과일:
    print(변수)

In [ ]:
사과
귤
수박 출력

#### 132. for문의 실행결과를 예측하라.

> 과일 = ["사과", "귤", "수박"]

> for 변수 in 과일:
>>  print("#####")

In [3]:
#####
#####
#####

#### 133.다음 for 문과 동일한 기능을 수행하는 코드를 작성하세요.

> for 변수 in ["A", "B", "C"]:
>>  print(변수)

In [7]:
word = ["A", "B", "C"]
for i in word :
    print(i)

A
B
C


#### 134. for문을 풀어서 동일한 동작을하는 코드를 작성하라.

> for 변수 in ["A", "B", "C"]:
>>  print("출력:", 변수)

In [9]:
word = ["A", "B", "C"]
print("출력 : ", word[0])
print("출력 : ", word[1])
print("출력 : ", word[2])

출력 :  A
출력 :  B
출력 :  C


In [11]:
word = ["A", "B", "C"]
for i in word :
    print("출력 : ", i)

출력 :  A
출력 :  B
출력 :  C


#### 135. for문을 풀어서 동일한 동작을 하는 코드를 작성하라.

> for 변수 in ["A", "B", "C"]:
>>  b = 변수.lower()

>>  print("변환:", b)

In [ ]:
word = ["A", "B", "C"]
print("출력 : ", word[0].lower())
print("출력 : ", word[1].lower())
print("출력 : ", word[2].lower())

In [13]:
word = ["A", "B", "C"]
for i in word :
    print("출력 : ", i.lower())

출력 :  a
출력 :  b
출력 :  c


In [17]:
word = ["A", "B", "C"]
for i in range(3) :
    print("출력 : ", word[i].lower())

출력 :  a
출력 :  b
출력 :  c


#### 136. 다음 코드를 for문으로 작성하라.

> 변수 = 10

> print(변수)

> 변수 = 20

> print(변수)

> 변수 = 30

> print(변수)


 



In [21]:
num = 0
for i in range(3) :
    num = int(input())
    print(num)

 10


10


 20


20


 30


30


In [23]:
for i in range(10, 40, 10) :
    print(i)

10
20
30


In [25]:
nums = [10, 20 ,30]
for i in range(3) :
    print(nums[i])

10
20
30


#### 137. 다음 코드를 for문으로 작성하라.

> print(10)

> print(20)

> print(30)



In [27]:
nums = [10, 20 ,30]
for i in range(3) :
    print(nums[i])

10
20
30


In [29]:
for i in range(10, 40, 10) :
    print(i)

10
20
30


In [31]:
num = 0
for i in range(3) :
    num = int(input())
    print(num)

 10


10


 20


20


 30


30


#### 138. 다음 코드를 for문으로 작성하라.

> print(10)

> print("-------")

> print(20)

> print("-------")

> print(30)

> print("-------")

In [33]:
for i in range(3) :
    nums = int(input())
    print(nums)
    print("--------")

 10


10
--------


 20


20
--------


 30


30
--------


In [35]:
nums = [10, 20 ,30]
for i in range(3) :
    print(nums[i])
    print("--------")

10
--------
20
--------
30
--------


#### 139. 다음 코드를 for문으로 작성하라.

- print("++++")
- print(10)
- print(20)
- print(30)

In [37]:
print("++++")
for i in nums :
    print(i)

++++
10
20
30


#### 140. 다음 코드를 for문으로 작성하라.

> print("-------")

> print("-------")

> print("-------")

> print("-------")

In [39]:
for i in range(4) :
    print("-------")

-------
-------
-------
-------


In [41]:
for i in range(2) :
    for j in range(2) :
        print("-------")

-------
-------
-------
-------


In [47]:
print("-------")

print("-------")

print("-------")

print("-------")

-------
-------
-------
-------


#### 141. 다음과 같이 판매가가 저장된 리스트가 있을 때 부가세가 포함된 가격을 for 문을 사용해서 화면에 출력하라. 단 부가세는 10원으로 가정한다.

> 리스트 = [100, 200, 300]

> 110

> 210

> 310

In [49]:
list = [100, 200, 300]
for i in range(3) :
    print(list[i] + 10)

110
210
310


In [63]:
list1 = [100, 200, 300]
for i in list1 :
    print(i + 10)

110
210
310


#### 142. for 문을 사용해서 리스트에 저장된 값을 다음과 같이 출력하라.

> 리스트 = ["김밥", "라면", "튀김"]

> 오늘의 메뉴: 김밥

> 오늘의 메뉴: 라면

> 오늘의 메뉴: 튀김

In [65]:
list1 = ["김밥", "라면", "튀김"]
for i in list1 :
    print("오늘의 메뉴: ", i)

오늘의 메뉴:  김밥
오늘의 메뉴:  라면
오늘의 메뉴:  튀김


In [67]:
list1 = ["김밥", "라면", "튀김"]
for i in range(len(list1)) :
    print("오늘의 메뉴: ", list1[i])

오늘의 메뉴:  김밥
오늘의 메뉴:  라면
오늘의 메뉴:  튀김


#### 143. 리스트에 주식 종목이름이 저장돼 있다.

- 리스트 = ["SK하이닉스", "삼성전자", "LG전자"]
- 저장된 문자열의 길이를 다음과 같이 출력하라.

> 6

> 4

> 4

In [71]:
list1 = ["SK하이닉스", "삼성전자", "LG전자"]
for i in range(3) :
    print(len(list1[i]))

6
4
4


In [73]:
list1 = ["SK하이닉스", "삼성전자", "LG전자"]
for i in list1 :
    print(len(i))

6
4
4


#### 144. 리스트에는 동물이름이 문자열로 저장돼 있다.

- 리스트 = ['dog', 'cat', 'parrot']
- 동물 이름과 글자수를 다음과 같이 출력하라.

> dog 3

> cat 3

> parrot 6

In [77]:
list1 = ['dog', 'cat', 'parrot']
for i in range(3) :
    print(list1[i], len(list1[i]))

dog 3
cat 3
parrot 6


#### 145. 리스트에 동물 이름 저장돼 있다.

- 리스트 = ['dog', 'cat', 'parrot']
- for문을 사용해서 동물 이름의 첫 글자만 출력하라.

> d

> c

> p

In [79]:
list1 = ['dog', 'cat', 'parrot']
for i in range(3) :
    print(list1[i][:1])

d
c
p


In [81]:
list1 = ['dog', 'cat', 'parrot']
for i in list1 :
    print(i[:1])

d
c
p


#### 146
- 리스트에는 세 개의 숫자가 바인딩돼 있다.

- 리스트 = [1, 2, 3]
- for문을 사용해서 다음과 같이 출력하라.

> 3 x 1

> 3 x 2

> 3 x 3

In [83]:
nums = [1, 2, 3]
for i in nums :
    print(f"3 X {i}")

3 X 1
3 X 2
3 X 3


In [89]:
nums = [1, 2, 3]
for i in nums :
    print("3 X %d" %i)

3 X 1
3 X 2
3 X 3


#### 147
- 리스트에는 세 개의 숫자가 바인딩돼 있다.

- 리스트 = [1, 2, 3]
- for문을 사용해서 다음과 같이 출력하라.

- 3 x 1 = 3
- 3 x 2 = 6
- 3 x 3 = 9

In [91]:
nums = [1, 2, 3]
mul = 0
for i in nums :
    mul = 3 * i
    print(f'3 X {i} = {mul}')

3 X 1 = 3
3 X 2 = 6
3 X 3 = 9


In [93]:
nums = [1, 2, 3]
mul = 0
for i in range(1, 4) :
    mul = 3 * i
    print(f'3 X {i} = {mul}')

3 X 1 = 3
3 X 2 = 6
3 X 3 = 9


In [95]:
nums = [1, 2, 3]
mul = 0
for i in range(1, len(nums) + 1):
    mul = 3 * i
    print(f'3 X {i} = {mul}')

3 X 1 = 3
3 X 2 = 6
3 X 3 = 9


#### 148. 리스트에는 네 개의 문자열이 바인딩돼 있다.

- 리스트 = ["가", "나", "다", "라"]
- for문을 사용해서 다음과 같이 출력하라.

- 나
- 다
- 라

In [97]:
words = ["가", "나", "다", "라"]
for i in range(1, len(words)) :
    print(words[i])

나
다
라


In [107]:
words = ["가", "나", "다", "라"]
for i in words :
    if i != '가' :
        print(i)

나
다
라


In [105]:
words = ["가", "나", "다", "라"]
for i in range(1, 4) :
    print(words[i])

나
다
라


#### 149. 리스트에는 네 개의 문자열이 바인딩돼 있다.

- 리스트 = ["가", "나", "다", "라"]
for문을 사용해서 다음과 같이 출력하라.

- 가
- 다

In [111]:
words = ["가", "나", "다", "라"]
for i in range(0, len(words), 2) :
    print(words[i])

가
다


#### 150. 리스트에는 네 개의 문자열이 바인딩돼 있다.

- 리스트 = ["가", "나", "다", "라"]
- for문을 사용해서 다음과 같이 출력하라.

- 라
- 다
- 나
- 가

In [117]:
words = ["가", "나", "다", "라"]
for i in range(len(words)) :
    words.sort(reverse=True)
    print(words[i])

라
다
나
가


In [119]:
words = ["가", "나", "다", "라"]
words.sort(reverse=True)
for i in words :
    print(i)

라
다
나
가


#### 151. 리스트에는 네 개의 정수가 저장돼 있다.

- 리스트 = [3, -20, -3, 44]
- for문을 사용해서 리스트의 음수를 출력하라.

> 20

> 3

In [121]:
numbers = [3, -20, -3, 44]
for i in range(4) :
    if numbers[i] < 0 :
        print( -1 * numbers[i])

20
3


In [123]:
numbers = [3, -20, -3, 44]
for i in numbers :
    if i < 0 :
        print(-1 * i)

20
3


#### 152. for문을 사용해서 3의 배수만을 출력하라.

- 리스트 = [3, 100, 23, 44]
> 3

In [125]:
num = [3, 100, 23, 44]
for i in num :
    if i % 3 == 0 :
        print(i)

3


#### 153. 리스트에서 20 보다 작은 3의 배수를 출력하라

- 리스트 = [13, 21, 12, 14, 30, 18]

> 12

> 18

In [127]:
nums = [13, 21, 12, 14, 30, 18]
for i in nums :
    if i < 20 :
        if i % 3 == 0:
            print(i)

12
18


In [129]:
nums = [13, 21, 12, 14, 30, 18]
for i in range(len(nums)) :
    if nums[i] < 20 :
        if nums[i] % 3 == 0:
            print(nums[i])

12
18


#### 154. 리스트에서 세 글자 이상의 문자를 화면에 출력하라

- 리스트 = ["I", "study", "python", "language", "!"]
- study
- python
- language

In [141]:
string1 = ["I", "study", "python", "language", "!"]
for i in range(len(string1)) :
    if len(string1[i]) >= 3 :
        print(string1[i])

study
python
language


#### 155. 리스트에서 대문자만 화면에 출력하라.

- 리스트 = ["A", "b", "c", "D"]
- A
- D
- (참고) isupper() 메서드는 대문자 여부를 판별합니다.

> 변수 = "A"

> 변수.isupper()

> True

> 변수 = "a"

> 변수.isupper()

> False

In [151]:
words = ["A", "b", "c", "D"]
for i in words :
    if i.isupper() == True :
        print(i)

A
D


#### 156. 리스트에서 소문자만 화면에 출력하라.

- 리스트 = ["A", "b", "c", "D"]
- b
- c

In [153]:
words = ["A", "b", "c", "D"]
for i in words :
    if i.islower() == True :
        print(i)

b
c


In [155]:
words = ["A", "b", "c", "D"]
for i in range(len(words)) :
    if words[i].islower() == True :
        print(words[i])

b
c


#### 157. 이름의 첫 글자를 대문자로 변경해서 출력하라.

- 리스트 = ['dog', 'cat', 'parrot']
- Dog
- Cat
- Parrot
- (참고) upper() 메서드는 문자열을 대문자로 변경합니다.

> 변수 = "a"

> a.upper()

> A

> 변수 = "abc"

> 변수.upper()

> ABC

In [165]:
animal = ['dog', 'cat', 'parrot']
for i in animal :
    print(i.capitalize())

Dog
Cat
Parrot


In [167]:
animal = ['dog', 'cat', 'parrot']
for i in range(len(animal)) :
    print(animal[i].capitalize())

Dog
Cat
Parrot


In [169]:
animal = ['dog', 'cat', 'parrot']
for i in range(3) :
    print(animal[i].capitalize())

Dog
Cat
Parrot


#### 158. 파일 이름이 저장된 리스트에서 확장자를 제거하고 파일 이름만 화면에 출력하라. (힌트: split() 메서드)

- 리스트 = ['hello.py', 'ex01.py', 'intro.hwp']
- hello
- ex01
- intro

In [175]:
file_name = ['hello.py', 'ex01.py', 'intro.hwp']
for i in file_name :
    name = i.split('.')[0]
    print(name)

hello
ex01
intro


In [177]:
file_name = ['hello.py', 'ex01.py', 'intro.hwp']
for i in range(3) :
    name = file_name[i].split('.')[0]
    print(name)

hello
ex01
intro


#### 159. 파일 이름이 저장된 리스트에서 확장자가 .h인 파일 이름을 출력하라.

- 리스트 = ['intra.h', 'intra.c', 'define.h', 'run.py']
- intra.h
- define.h

In [184]:
file_name = ['intra.h', 'intra.c', 'define.h', 'run.py']
for i in file_name :
    name = i.split('.')[1]
    if name == 'h' :
        print(i)

intra.h
define.h


#### 160. 파일 이름이 저장된 리스트에서 확장자가 .h나 .c인 파일을 화면에 출력하라.

- 리스트 = ['intra.h', 'intra.c', 'define.h', 'run.py']
- intra.h
- intra.c
- define.h

In [186]:
file_name = ['intra.h', 'intra.c', 'define.h', 'run.py']
for i in file_name :
    name = i.split('.')[1]
    if name == 'h' :
        print(i)
    elif name == 'c' :
        print(i)

intra.h
intra.c
define.h


In [190]:
file_name = ['intra.h', 'intra.c', 'define.h', 'run.py']
for i in range(len(file_name)) :
    name = file_name[i].split('.')[1]
    if name == 'h' :
        print(file_name[i])
    elif name == 'c' :
        print(file_name[i])

intra.h
intra.c
define.h


#### 161. for문과 range 구문을 사용해서 0~99까지 한 라인에 하나씩 순차적으로 출력하는 프로그램을 작성하라.

In [188]:
for i in range(100) :
    print(i)

0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99


#### 162. 월드컵은 4년에 한 번 개최된다. range()를 사용하여 2002~2050년까지 중 월드컵이 개최되는 연도를 출력하라.

- 2002
- 2006
- 2010
- ...
- 2042
- 2046
- 2050
- 참고) range의 세번 째 파라미터는 증감폭을 결정합니다.

> print(list(range(0, 10, 2)))

> [0, 2, 4, 6, 8]

In [194]:
for i in range(2002, 2054, 4) :
    print(i)

2002
2006
2010
2014
2018
2022
2026
2030
2034
2038
2042
2046
2050


#### 163. 1부터 30까지의 숫자 중 3의 배수를 출력하라.

- 3 
- 6 
- 9 
- 12 
- 15 
- 18 
- 21 
- 24 
- 27 
- 30

In [198]:
for i in range(1, 31) :
    if i % 3 == 0 :
        print(i)

3
6
9
12
15
18
21
24
27
30


#### 164. 99부터 0까지 1씩 감소하는 숫자들을, 한 라인에 하나씩 출력하라.

In [202]:
for i in range(99, -1, -1) :
    print(i)

99
98
97
96
95
94
93
92
91
90
89
88
87
86
85
84
83
82
81
80
79
78
77
76
75
74
73
72
71
70
69
68
67
66
65
64
63
62
61
60
59
58
57
56
55
54
53
52
51
50
49
48
47
46
45
44
43
42
41
40
39
38
37
36
35
34
33
32
31
30
29
28
27
26
25
24
23
22
21
20
19
18
17
16
15
14
13
12
11
10
9
8
7
6
5
4
3
2
1
0


#### 165. for문을 사용해서 아래와 같이 출력하라.

- 0.0
- 0.1
- 0.2
- 0.3
- 0.4
- 0.5
- ...
- 0.9

In [206]:
for i in range(10) :
    print(f'0.{i}')

0.0
0.1
0.2
0.3
0.4
0.5
0.6
0.7
0.8
0.9


#### 166. 구구단 3단을 출력하라.

- 3x1 = 3
- 3x2 = 6
- 3x3 = 9
- 3x4 = 12
- 3x5 = 15
- 3x6 = 18
- 3x7 = 21
- 3x8 = 24
- 3x9 = 27

In [208]:
for i in range(1, 10) :
    mul = 3 * i
    print(f'3X{i} = {mul}')

3X1 = 3
3X2 = 6
3X3 = 9
3X4 = 12
3X5 = 15
3X6 = 18
3X7 = 21
3X8 = 24
3X9 = 27


#### 167. 구구단 3단을 출력하라. 단 홀수 번째만 출력한다.

- 3x1 = 3
- 3x3 = 9
- 3x5 = 15
- 3x7 = 21
- 3x9 = 27

In [210]:
for i in range(1, 10) :
    mul = 3 * i
    if i % 2 == 1 :
        print(f'3X{i} = {mul}')

3X1 = 3
3X3 = 9
3X5 = 15
3X7 = 21
3X9 = 27


In [212]:
for i in range(1, 10, 2) :
    mul = 3 * i
    print(f'3X{i} = {mul}')

3X1 = 3
3X3 = 9
3X5 = 15
3X7 = 21
3X9 = 27


#### 168. 1~10까지의 숫자에 대해 모두 더한 값을 출력하는 프로그램을 for 문을 사용하여 작성하라.

- 합 : 55

In [214]:
sum = 0
for i in range(11) :
    sum += i
print(sum)

55


In [216]:
sum = 0
for i in range(1, 11) :
    sum += i
print(sum)

55


#### 169. 1~10까지의 숫자 중 모든 홀수의 합을 출력하는 프로그램을 for 문을 사용하여 작성하라.

- 합: 25

In [218]:
odd_sum = 0
for i in range(1, 11, 2) :
    odd_sum += i
print(odd_sum)

25


#### 170. 1~10까지의 숫자를 모두 곱한 값을 출력하는 프로그램을 for 문을 사용하여 작성하라.

In [220]:
mul = 1
for i in range(1, 11) :
    mul *= i
    print(mul)

1
2
6
24
120
720
5040
40320
362880
3628800


#### 171. 아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.

- price_list = [32100, 32150, 32000, 32500]
- 32100
- 32150
- 32000
- 32500

In [222]:
price_list = [32100, 32150, 32000, 32500]
for i in range(len(price_list)) :
    print(price_list[i])

32100
32150
32000
32500


In [224]:
price_list = [32100, 32150, 32000, 32500]
for i in range(4) :
    print(price_list[i])

32100
32150
32000
32500


#### 172. 아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.

- price_list = [32100, 32150, 32000, 32500]
- 0 32100
- 1 32150
- 2 32000
- 3 32500

In [226]:
price_list = [32100, 32150, 32000, 32500]
for i in range(4) :
    print(i, price_list[i])

0 32100
1 32150
2 32000
3 32500


#### 173. 아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.

- price_list = [32100, 32150, 32000, 32500]
- 3 32100
- 2 32150
- 1 32000
- 0 32500

In [234]:
price_list = [32100, 32150, 32000, 32500]
for i in range(3, -1, -1) :
    print(i, price_list[3-i])

3 32100
2 32150
1 32000
0 32500


In [240]:
price_list = [32100, 32150, 32000, 32500]
for i in range(4) :
    print(3-i, price_list[i])

3 32100
2 32150
1 32000
0 32500


#### 174. 아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.

- price_list = [32100, 32150, 32000, 32500]
- 100 32150
- 110 32000
- 120 32500

In [242]:
price_list = [32100, 32150, 32000, 32500]
for i in range(1, 4) :
    print(100 + (i-1) * 10, price_list[i])

100 32150
110 32000
120 32500


#### 175. my_list를 아래와 같이 출력하라.

- my_list = ["가", "나", "다", "라"]
- 가 나
- 나 다
- 다 라

In [250]:
my_list = ["가", "나", "다", "라"]
for i in range(3) :
    print(my_list[i], my_list[i+1])

가 나
나 다
다 라


In [258]:
my_list = ["가", "나", "다", "라"]
for i in range(3) :
    print(my_list[i:i+2])

['가', '나']
['나', '다']
['다', '라']


#### 176. 리스트를 아래와 같이 출력하라.

- my_list = ["가", "나", "다", "라", "마"]
- 가 나 다
- 나 다 라
- 다 라 마

In [252]:
my_list = ["가", "나", "다", "라", "마"]
for i in range(3) :
    print(my_list[i], my_list[i+1], my_list[i+2])

가 나 다
나 다 라
다 라 마


In [260]:
my_list = ["가", "나", "다", "라", "마"]
for i in range(3) :
    print(my_list[i:i+3])

['가', '나', '다']
['나', '다', '라']
['다', '라', '마']


#### 177. 복문과 range 함수를 사용해서 my_list를 아래와 같이 출력하라.

- my_list = ["가", "나", "다", "라"]
- 라 다
- 다 나
- 나 가

In [269]:
my_list = ["가", "나", "다", "라"]
for i in range(3) :
    print(my_list[3-i], my_list[2-i])

라 다
다 나
나 가


#### 178. 리스트에는 네 개의 정수가 저장되어 있다. 각각의 데이터에 대해서 자신과 우측값과의 차분값을 화면에 출력하라.

- my_list = [100, 200, 400, 800]
- 예를들어 100을 기준으로 우측에 위치한 200과의 차분 값를 화면에 출력하고, 200을 기준으로 우측에 위치한 400과의 차분값을 화면에 출력한다. 이어서 400을 기준으로 우측에 위치한 800과의 차분값을 화면에 출력한다.

- 100
- 200
- 400

In [300]:
my_list = [100, 200, 400, 800]
for i in range(3) :
    print(my_list[i+1] - my_list[i])

100
200
400


In [306]:
my_list = [100, 200, 400, 800]
for i in range(3) :
    print(my_list[-3+i] - my_list[-4+i])

100
200
400


#### 179. 리스트에는 6일 간의 종가 데이터가 저장되어 있다. 종가 데이터의 3일 이동 평균을 계산하고 이를 화면에 출력하라.

- my_list = [100, 200, 400, 800, 1000, 1300]
- 첫 번째 줄에는 100, 200, 400의 평균값이 출력된다. 두 번째 줄에는 200, 400, 800의 평균값이 출력된다. 같은 방식으로 나머지 데이터의 평균을 출력한다.

- 233.33333333333334
- 466.6666666666667
- 733.3333333333334
- 1033.3333333333333

In [318]:
my_list = [100, 200, 400, 800, 1000, 1300]
for i in range(4) :
    sum = my_list[i] + my_list[i+1] + my_list[i+2]
    avg = sum / 3
    print(avg)

233.33333333333334
466.6666666666667
733.3333333333334
1033.3333333333333


In [322]:
my_list = [100, 200, 400, 800, 1000, 1300]
for i in range(4) :
    sum = my_list[5-i] + my_list[4-i] + my_list[3-i]
    avg = sum / 3
    print(avg)

1033.3333333333333
733.3333333333334
466.6666666666667
233.33333333333334


#### 180. 리스트에 5일간의 저가, 고가 정보가 저장돼 있다. 고가와 저가의 차를 변동폭이라고 정의할 때, low, high 두 개의 리스트를 사용해서 5일간의 변동폭을 volatility 리스트에 저장하라.

- low_prices  = [100, 200, 400, 800, 1000]
- high_prices = [150, 300, 430, 880, 1000]

In [324]:
low_prices = [100, 200, 400, 800, 1000]
high_prices = [150, 300, 430, 880, 1000]
volatility = []

for i in range(len(low_prices)) :
    volatility.append(high_prices[i] - low_prices[i])
print(volatility)

[50, 100, 30, 80, 0]


#### 181. 아래 표에서 하나의 행을 하나의 리스트로, 총 3개의 리스트를 갖는 이차원 리스트 apart를 정의하라.

- 101호 |	102호
- 201호 |	202호
- 301호 |	302호

In [328]:
apart1 = ['101호', '102호']
apart2 = ['201호', '202호']
apart3 = ['301호', '302호']
apart = [apart1, apart2, apart3]
print(apart)

[['101호', '102호'], ['201호', '202호'], ['301호', '302호']]


#### 182. 아래 표에서 하나의 열을 하나의 리스트로, 총 2개의 리스트를 갖는 이차원 리스트 stock을 정의하라.

- 시가	종가
- 100	80
- 200	210
- 300	330

In [330]:
stock1 = [100, 80]
stock2 = [200, 210]
stock3 = [300, 330]
stock = [stock1, stock2, stock3]
print(stock)

[[100, 80], [200, 210], [300, 330]]


#### 183. 아래 표를 stock 이름의 딕셔너리로 표현하라.시가를 key로 저장하고, 나머지 같은 열의 데이터를 리스트로 저장해서 value로 저장한다. 종가 역시 key로 저장하고 나머지 같은 열의 데이터를 리스트로 저장해서 value로 저장한다.

- 시가	종가
- 100	80
- 200	210
- 300	330

In [334]:
stock = {'시가' : [100, 200, 300], '종가' : [80, 210, 330] }
print(stock)

{'시가': [100, 200, 300], '종가': [80, 210, 330]}


#### 184. 아래 표를 stock 이라는 이름의 딕셔너리로 표현하라. 날짜를 key로 저장하고, 나머지 같은 행의 데이터를 리스트로 저장해서 value로 저장한다. 첫 열이 날짜이다.

- 10/10	80	110	70	90
- 10/11	210	230	190	200

In [336]:
stock = {'10/10' : [80, 110, 70, 90], '10/11':[210, 230, 190, 200]}
print(stock)

{'10/10': [80, 110, 70, 90], '10/11': [210, 230, 190, 200]}


#### 185. 리스트에 저장된 데이터를 아래와 같이 출력하라.

- apart = [ [101, 102], [201, 202], [301, 302] ]
- 101 호
- 102 호
- 201 호
- 202 호
- 301 호
- 302 호

In [342]:
for i in range(3) :
    for j in range(2) :
        print(apart[i][j])

101호
102호
201호
202호
301호
302호


#### 186. 리스트에 저장된 데이터를 아래와 같이 출력하라.

- apart = [ [101, 102], [201, 202], [301, 302] ]
- 301 호
- 302 호
- 201 호
- 202 호
- 101 호
- 102 호

In [346]:
for i in range(3) :
    for j in range(2) :
        print(apart[2-i][j])

301호
302호
201호
202호
101호
102호


#### 187. 리스트에 저장된 데이터를 아래와 같이 출력하라.

- apart = [ [101, 102], [201, 202], [301, 302] ]
- 302 호
- 301 호
- 202 호
- 201 호
- 102 호
- 101 호

In [348]:
for i in range(3) :
    for j in range(2) :
        print(apart[2-i][1-j])

302호
301호
202호
201호
102호
101호


#### 188. 리스트에 저장된 데이터를 아래와 같이 출력하라.

- apart = [ [101, 102], [201, 202], [301, 302] ]
- 101 호
- -----
- 102 호
- -----
- 201 호
- -----
- 202 호
- -----
- 301 호
- -----
- 302 호
- -----

In [350]:
for i in range(3) :
    for j in range(2) :
        print(apart[i][j])
        print('-----')

101호
-----
102호
-----
201호
-----
202호
-----
301호
-----
302호
-----


#### 189. 리스트에 저장된 데이터를 아래와 같이 출력하라.

- apart = [ [101, 102], [201, 202], [301, 302] ]
- 101 호
- 102 호
- -----
- 201 호
- 202 호
- -----
- 301 호
- 302 호
- -----

In [352]:
for i in range(3) :
    for j in range(2) :
        print(apart[i][j])
    print('-----')

101호
102호
-----
201호
202호
-----
301호
302호
-----


#### 190. 리스트에 저장된 데이터를 아래와 같이 출력하라.

- apart = [ [101, 102], [201, 202], [301, 302] ]
- 101 호
- 102 호
- 201 호
- 202 호
- 301 호
- 302 호
- -----

In [354]:
for i in range(3) :
    for j in range(2) :
        print(apart[i][j])
print('-----')

101호
102호
201호
202호
301호
302호
-----


#### 191. data에는 매수한 종목들의 OHLC (open/high/low/close) 가격 정보가 바인딩 되어있다.

data = [
    [ 2000,  3050,  2050,  1980],
    [ 7500,  2050,  2050,  1980],
    [15450, 15050, 15550, 14900]
]
#### 수수료를 0.014 %로 가정할 때, 각 가격에 수수료를 포함한 가격을 한라인에 하나씩 출력하라.

- 2000.28
- 3050.427
- 2050.2870000000003
- ...

In [360]:
data = [ [ 2000, 3050, 2050, 1980], [ 7500, 2050, 2050, 1980], [15450, 15050, 15550, 14900] ]
for i in range(3) :
    for j in range(4) :
        su = (data[i][j] * 0.014 / 100)
        print(data[i][j] + su)

2000.28
3050.427
2050.287
1980.2772
7501.05
2050.287
2050.287
1980.2772
15452.163
15052.107
15552.177
14902.086


#### 192. 191번의 출력 결과에 행단위로 "----" 구분자를 추가하라.

- 2000.28
- 3050.427
- 2050.2870000000003
- 1980.2772
- ----
- 7501.05
- 2050.2870000000003
- 2050.2870000000003
- 1980.2772
- ----
- 15452.163
- 15052.107
- 15552.177
- 14902.086000000001
- ----

In [362]:
data = [ [ 2000, 3050, 2050, 1980], [ 7500, 2050, 2050, 1980], [15450, 15050, 15550, 14900] ]
for i in range(3) :
    for j in range(4) :
        su = (data[i][j] * 0.014 / 100)
        print(data[i][j] + su)
    print('-----')

2000.28
3050.427
2050.287
1980.2772
-----
7501.05
2050.287
2050.287
1980.2772
-----
15452.163
15052.107
15552.177
14902.086
-----


#### 193. 192 번 문제의 결과값을 result 이름의 리스트에 1차원 배열로 저장하라.

> print(result)

> [2000.28, 3050.427, 2050.2870000000003, 1980.2772, 7501.05, 2050.2870000000003, 2050.2870000000003, ...]

In [366]:
data = [ [ 2000, 3050, 2050, 1980], [ 7500, 2050, 2050, 1980], [15450, 15050, 15550, 14900] ]
result = []
for i in range(3) :
    for j in range(4) :
        su = (data[i][j] * 0.014 / 100)
        result.append(data[i][j] + su)
print(result)

[2000.28, 3050.427, 2050.287, 1980.2772, 7501.05, 2050.287, 2050.287, 1980.2772, 15452.163, 15052.107, 15552.177, 14902.086]


#### 194. 191번 문제의 결괏값을 result 이름의 리스트에 2차원 배열로 저장하라. 저장 포맷은 아래와 같다. 각 행에 대한 데이터끼리 리스트에 저장되어야 한다.

> print(result)
- [
 [2000.28, 3050.427, 2050.2870000000003, 1980.2772],
 [7501.05, 2050.2870000000003, 2050.2870000000003, 1980.2772],
 [15452.163, 15052.107, 15552.177, 14902.086000000001]
]

In [38]:
data = [ [ 2000, 3050, 2050, 1980], [ 7500, 2050, 2050, 1980], [15450, 15050, 15550, 14900] ]
result = []
list_result = []
su = 0
for i in range(3) :
    for j in range(4) :
        su = (data[i][j] * 0.014 / 100)
        list_result.append(data[i][j] + su)

result = [list_result[i:i+4] for i in range(0, len(list_result), 4)]
print(result)

[[2000.28, 3050.427, 2050.287, 1980.2772], [7501.05, 2050.287, 2050.287, 1980.2772], [15452.163, 15052.107, 15552.177, 14902.086]]


#### 195. ohlc 리스트에는 시가(open), 고가 (high), 저가 (low) , 종가(close)가 날짜별로 저장돼 있다. 화면에 종가데이터를 출력하라.

- ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
- 100
- 190
- 310

In [40]:
ohlc = [["open", "high", "low", "close"], [100, 110, 70, 100], [200, 210, 180, 190], [300, 310, 300, 310]]
for i in range(1, 4) :
    print(ohlc[i][3])

100
190
310


#### 196. ohlc 리스트에는 시가(open), 고가 (high), 저가 (low) , 종가(close)가 날짜별로 저장돼 있다. 종가가 150원보다 큰경우에만 종가를 출력하라.

- ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
- 190
- 310

In [46]:
ohlc = [["open", "high", "low", "close"], [100, 110, 70, 100], [200, 210, 180, 190], [300, 310, 300, 310]]
for i in range(1, 4) :
    if ohlc[i][3] > 150 :
        print(ohlc[i][3])

190
310


#### 197. ohlc 리스트에는 시가(open), 고가 (high), 저가 (low) , 종가(close)가 날짜별로 저장돼 있다. 종가가 시가 보다 크거나 같은 경우에만 종가를 출력하라.

- ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
- 100
- 310

In [50]:
ohlc = [["open", "high", "low", "close"], [100, 110, 70, 100], [200, 210, 180, 190], [300, 310, 300, 310]]
for i in range(1, 4) :
    if ohlc[i][0] <= ohlc[i][3] :
        print(ohlc[i][3])

100
310


#### 198. ohlc 리스트에는 시가(open), 고가 (high), 저가 (low) , 종가(close)가 날짜별로 저장돼 있다. 고가와 저가의 차이를 변동폭으로 정의할 때 변동폭을 volatility 이름의 리스트에 저장하라.

> ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]

> print(volatility)

>[40, 30, 10]

In [52]:
ohlc = [["open", "high", "low", "close"], [100, 110, 70, 100], [200, 210, 180, 190], [300, 310, 300, 310]]
volatility = []
for i in range(1, 4) :
    volatility.append(ohlc[i][1] - ohlc[i][2])
print(volatility)

[40, 30, 10]


In [58]:
ohlc = [["open", "high", "low", "close"], [100, 110, 70, 100], [200, 210, 180, 190], [300, 310, 300, 310]]
volatility = []
i = 1
while i < 4:
    volatility.append(ohlc[i][1] - ohlc[i][2])
    i += 1
print(volatility)

[40, 30, 10]


#### 199. 리스트에는 3일 간의 ohlc 데이터가 저장돼 있다. 종가가 시가보다 높은 날의 변동성 (고가 - 저가)을 화면에 출력하라.

- ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
- 종가가 시가보다 높은 거래일의 OHLC는 [300, 310, 300, 310] 이다. 따라서 이 거래일의 변동성은 10 (310 - 300)이다.

- 10

In [60]:
ohlc = [["open", "high", "low", "close"], [100, 110, 70, 100], [200, 210, 180, 190], [300, 310, 300, 310]]
for i in range(1, 4) :
    if ohlc[i][0] < ohlc[i][3] :
        print(ohlc[i][1] - ohlc[i][2])

10


#### 200. 리스트에는 3일 간의 ohlc 데이터가 저장돼 있다. 시가에 매수해서 종가에 매도 했을 경우 총 수익금을 계산하라.

- ohlc = [["open", "high", "low", "close"],
        [100, 110, 70, 100],
        [200, 210, 180, 190],
        [300, 310, 300, 310]]
- 1일차 수익 0원 (100 - 100)
- 2일차 수익 -10원 (190 - 200)
- 3일차 수익 10원 (310 - 300) 이다.

- 0

In [64]:
ohlc = [["open", "high", "low", "close"], [100, 110, 70, 100], [200, 210, 180, 190], [300, 310, 300, 310]]
ohlc_revenue = 0
for i in range(1, 4) :
    ohlc_revenue += ohlc[i][3] - ohlc[i][0]
print(ohlc_revenue)

0
